cabal.project

```diff
@@ -176,12 +168,17 @@ constraints:
   , cardano-api ^>=8.2
   , cardano-slotting >= 0.1
   , ouroboros-network ^>= 0.8.1.0
+  , cardano-cli == 8.1.2
+
+  -- `cardano-cli-8.1.2` uses `optparse-applicative-fork-0.16.2.0`
+  -- and `cardano-wallet` uses `cardano-cli` as a dependency, so
+  -- it needs to use the same version of `optparse-applicative-fork`.
+  -- Otherwise, we'd use the latest version of `optparse-applicative`.
+  , optparse-applicative-fork == 0.16.2.0
 
```

Comment:
```suggestion

  -- Pin version due to a significant change in the public API
  -- compared to older version (incompatible pretty-printer).
  , optparse-applicative ^>= 0.18.1.0
```

I suggest to pin the version here, just to be explicit. 🤔 This will convert obscure compile errors into version bound clashes. For example, `bech32-1.1.4` claims to be compatible with `optparse-applicative-0.17.1.0`, but it's not. 🙈

Reply:
Fair enough.

On the other hand, `cardano-wallet` is currently a monorepo, so a global pin would be in line with the current style.